breadbox/breadbox/crud/dataset.py

```diff
@@ -1029,6 +1029,36 @@ def get_dataset_sample_labels_by_id(
         return {sample.given_id: sample.given_id for sample in samples}
 
 
+def get_dimension_labels_by_id(
+    db: SessionWithUser, dimension_type_name: str
+) -> dict[str, str]:
+    """
+    For a given dimension, get all IDs and labels that exist in the metadata.
+    """
+    dimension_type: Optional[DimensionType] = db.query(DimensionType).filter(
+        DimensionType.name == dimension_type_name
+    ).one_or_none()
```

Comment:
Oh thank you, that's a good suggestion. I just fixed it by moving this whole `get_dimension_labels_by_id` function into `crud/types.py`, so it can access `get_dimension_type`.  

While I was working on that, I also found about 4 other paces that _could_ be using this same `get_dimension_type` function instead re-implementing the query. However, I can't easily update them without creating a circular import (by importing `crud/types.py` into `crud/dataset.py`).

This is one of the reasons I would really like to make these crud files smaller and move a few things like label-loading into a service layer.